py3status/modules/systemd_suspend_inhibitor.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""
+Turn on and off systemd suspend inhibitor.
+
+Configuration parameters:
+    format: display format for this module
+        (default '[\?color=state SUSPEND [\?if=state OFF|ON]]')
+    lock_types: specify state to inhibit, comma separated list
+        https://www.freedesktop.org/wiki/Software/systemd/inhibit/
+        (default 'handle-lid-switch,idle,sleep')
+    thresholds: specify color thresholds to use
+        (default [(True, 'bad'), (False, 'good')])
+
+Format placeholders:
+    {state} systemd suspend inhibitor state, eg True, False
+
+Color thresholds:
+    xxx: print a color based on the value of `xxx` placeholder
+```
+
+
+Examples:
+# display SUSPEND ON/OFF
+systemd_suspend_inhibitor {
+    format = '[\?color=state SUSPEND [\?if=state OFF|ON]]'
+    thresholds = [(True, "bad"), (False, "good")]
+}
+```
+
+@author Cyrinux https://github.com/cyrinux
+@license BSD
+
+SAMPLE OUTPUT
+[{'full_text': 'SUSPEND ON', 'color': '#00FF00'}]
+
+off
+[{'full_text': 'SUSPEND OFF', 'color': '#FF0000'}]
+"""
+
+from dbus import SystemBus
+from os import close
+
+
+STRING_DBUS_EXCEPTION = "DBUS error, systemd-logind not started?"
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    format = "[\?color=state SUSPEND [\?if=state OFF|ON]]"
+    lock_types = "handle-lid-switch,idle,sleep"
+    thresholds = [(True, "bad"), (False, "good")]
+
+    def post_config_hook(self):
+        bus = SystemBus()
+        self.lock = None
+        self.lock_types = self.lock_types.replace(" ", "").replace(",", ":")
+        try:
+            self.login1 = bus.get_object(
+                "org.freedesktop.login1", "/org/freedesktop/login1"
+            )
+        except Exception:
+            raise Exception(STRING_DBUS_EXCEPTION)
+        self.thresholds_init = self.py3.get_color_names_list(self.format)
+
+    def systemd_suspend_inhibitor(self):
+        data = {"state": bool(self.lock)}
+
+        for x in self.thresholds_init:
+            if x in data:
+                self.py3.threshold_get_color(data[x], x)
+
+        return {
+            "cached_until": self.py3.CACHE_FOREVER,
+            "full_text": self.py3.safe_format(self.format, data),
+        }
+
+    def on_click(self, event):
+        if self.lock is None:
+            self.lock = self.login1.Inhibit(
+                self.lock_types,
+                "Py3Status",
+                "Systemd suspend inhibitor module",
+                "block",
+                dbus_interface="org.freedesktop.login1.Manager",
+            ).take()
+        else:
+            close(self.lock)
+            self.lock = None
+
+
+if __name__ == "__main__":
+    """
+    Run module in test mode.
+    """
+    from py3status.module_test import module_test
+
+    module_test(Py3status)
```

Comment:
Guess we can't test this in the terminal or with instances? They all run on its own... making it hard to tell if it's actually doing anything. Is there a better solution? If not, oh well.

Reply:
My way to check is running `watch -n1 systemd-inhibit | grep Py3`

Reply:
What should i do ?

Reply:
Do what you want. According to https://www.freedesktop.org/wiki/Software/systemd/logind/, you should be able to detect Py3status in **`ListInhibitors()`** `lists all currently active inhibitors. Returns an array of structures consisting of what, who, why, mode, user ID and process ID.`

I'm not sure how it would look, but I think checking for `Py3status` instead of `bool(self.lock)` would make the `state` more consistent across instances + test.... and if it turns to be unlocked or locked, assign the new/same value anyway.

Also, you'd want to use `cache_timeout` too to refresh instances.  All of this looks to be excessive just to make instances work... so yeah, I dunno what you should do too. Approved.

Reply:
Thanks for the review lasers, I'm agree that is excessive. I think we can start with this and iterate later. ;)